src/imagery/i.eodag/testsuite/test_eodag.py

```diff
@@ -0,0 +1,313 @@
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery <hamedashraf2004 gmail.com>
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Tests i.eodag input parsing / searching results.
+#              Uses NC Full data set.
+#
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.gutils import get_current_mapset, is_map_in_mapset
+from grass.exceptions import CalledModuleError
+from grass.pygrass.modules import Module
+from subprocess import PIPE
+
+
+class TestEodag(TestCase):
+    """When testing searching, we are using peps when testing because it gives
+    consistent results, unlike cop_dataspace & creodias.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Use temporary region settings"""
+        print("EHRERKJELJZD")
+        cls.use_temp_region()
+        cls.runModule(
+            "v.extract",
+            input="urbanarea",
+            where="NAME = 'Durham'",
+            output="durham",
+            overwrite=True,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete temporary region settings"""
+        cls.del_temp_region()
+
+    def test_search_without_date(self):
+        """Test"""
+        self.assertModule(
+            "i.eodag", flags="l", producttype="S2_MSI_L2A", clouds=30, quiet=True
+        )
+
+    def test_search_S2_MSI_L2A_default(self):
+        """Test"""
+        output = r"""S2A_MSIL2A_20240113T160621_N0510_R097_T17SPV_20240113T202049 2024-01-13T16:06:21  4% S2MSI2A
+S2A_MSIL2A_20240113T160621_N0510_R097_T17SPA_20240113T202049 2024-01-13T16:06:21  7% S2MSI2A
+S2B_MSIL2A_20240108T160639_N0510_R097_T17SPA_20240108T203022 2024-01-08T16:06:39 12% S2MSI2A
+S2B_MSIL2A_20240118T160609_N0510_R097_T17SPV_20240118T203136 2024-01-18T16:06:09 22% S2MSI2A"""
+        # provider used shall be creodias by default
+        i_eodag = Module(
+            "i.eodag",
+            flags="l",
+            producttype="S2_MSI_L2A",
+            clouds=30,
+            map="durham",
+            start="2024-01-01",
+            end="2024-02-01",
+            quiet=True,
+            stdout_=PIPE,
+        )
+        self.assertEqual(i_eodag.outputs["stdout"].value.strip(), output)
```

Comment:
If we have too strict test-criteria (here _assertEqual_ for the entire output, or an md5 test for the output file, e.g. reprocessing of the products with a new processing baseline would cause the test to fail. I would probably rather test if the requested product type is returned, the sensing time is within _start_ and _end_  and stuff like that...

Reply:
yes, just in case something change on their side (highly likely  based on 10+ years of the Sentinel data being out), I'd check if the time is within start and end, and if the product starts with S2 or so... Question is what happens with these tests if the site is down?? They will all fail and it is not a problem of the tool but on the server side...

Reply:
> Question is what happens with these tests if the site is down?? They will all fail and it is not a problem of the tool but on the server side...

If we unify all the tests to use "peps" as the provider, maybe we can run a pretty basic test of the module in the `setUpClass` method, and then save whether this test fails or not in flag. If it fails, then we can always return success in all of the tests, and if it passes we run the tests normally.